src/storage/disk_table.cc

```diff
@@ -850,6 +850,237 @@ void DiskTableTraverseIterator::NextPK() {
+    return it_->Valid();
+}
+
+const hybridse::codec::Row DiskTableKeyIterator::GetKey() {
+    hybridse::codec::Row row(
+        ::hybridse::base::RefCountedSlice::Create(pk_.c_str(), pk_.size()));
+    return row;
+}
+
+std::unique_ptr<::hybridse::vm::RowIterator> DiskTableKeyIterator::GetValue() {
+    std::unique_ptr<DiskTableRowIterator> wit(new DiskTableRowIterator(db_, it_, snapshot_, ttl_type_, expire_time_,
+                                                                       expire_cnt_, pk_, ts_, has_ts_idx_, ts_idx_));
+    return wit;
+}
+
+::hybridse::vm::RowIterator* DiskTableKeyIterator::GetRawValue() {
+    return new DiskTableRowIterator(db_, it_, snapshot_, ttl_type_, expire_time_, expire_cnt_, pk_, ts_, has_ts_idx_,
+                                    ts_idx_);
+}
+
+DiskTableRowIterator::DiskTableRowIterator(rocksdb::DB* db, rocksdb::Iterator* it, const rocksdb::Snapshot* snapshot,
+                                           ::openmldb::storage::TTLType ttl_type, uint64_t expire_time,
+                                           uint64_t expire_cnt, std::string pk, uint64_t ts, bool has_ts_idx,
+                                           uint32_t ts_idx)
+    : it_(it),
+      record_idx_(1),
+      expire_value_(expire_time, expire_cnt, ttl_type),
+      row_(),
+      pk_(pk),
+      row_pk_(pk),
+      ts_(ts),
+      has_ts_idx_(has_ts_idx),
+      ts_idx_(ts_idx) {}
+
+DiskTableRowIterator::~DiskTableRowIterator() {}
+
+bool DiskTableRowIterator::Valid() const {
+    if (!pk_valid_) return false;
+    if (!it_->Valid() || expire_value_.IsExpired(ts_, record_idx_)) {
+        return false;
+    }
+    return true;
+}
+
+void DiskTableRowIterator::Next() {
+    for (it_->Next(); it_->Valid(); it_->Next()) {
+        uint32_t cur_ts_idx = UINT32_MAX;
+        ParseKeyAndTs(has_ts_idx_, it_->key(), pk_, ts_, cur_ts_idx);
+        if (row_pk_ == pk_) {
+            if (has_ts_idx_ && (cur_ts_idx != ts_idx_)) {
+                continue;
+            }
+            record_idx_++;
+            pk_valid_ = true;
+        } else {
+            pk_valid_ = false;
+        }
+        break;
+    }
+}
+
+inline const uint64_t& DiskTableRowIterator::GetKey() const { return ts_; }
+
+const ::hybridse::codec::Row& DiskTableRowIterator::GetValue() {
+    rocksdb::Slice value = it_->value();
+    row_.Reset(reinterpret_cast<const int8_t*>(value.data()), value.size());
+    return row_;
+}
+
+void DiskTableRowIterator::Seek(const uint64_t& key) {
+    std::string combine;
+    uint64_t tmp_ts = key;
+    if (has_ts_idx_) {
+        combine = CombineKeyTs(row_pk_, tmp_ts, ts_idx_);
+    } else {
+        combine = CombineKeyTs(row_pk_, tmp_ts);
+    }
+    it_->Seek(rocksdb::Slice(combine));
+    for (; it_->Valid(); it_->Next()) {
+        uint32_t cur_ts_idx = UINT32_MAX;
+        ParseKeyAndTs(has_ts_idx_, it_->key(), pk_, ts_, cur_ts_idx);
+        if (pk_ == row_pk_) {
+            if (has_ts_idx_ && (cur_ts_idx != ts_idx_)) {
```

Comment:
why judge the ts_idx here

Reply:
`CombineKey` will combine `pk, ts_col, ts`. So here need to check if `ts_col` corresponds to this iterator.